binary_search_tree/tree.py

```diff
@@ -14,36 +14,127 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log n)
+    # Space Complexity: O(n)
```

Comment:
🪐 You aren't creating any new data structures here or holding recursive calls in memory, so space complexity would be O(1)